crates/ide_assists/src/handlers/extract_module.rs

```diff
@@ -0,0 +1,660 @@
+                            }
+                        },
+                        ast::Enum(it) => {
+                            let node_def = {
+                                let nod = ctx.sema.to_def(&it)?;
+                                Definition::ModuleDef(nod.into())
+                            };
+
+                            self.get_and_process_usages(ctx, builder, node_def);
+                            vis_node_occurences.push(it.syntax().clone());
+
+                            for desc in it.syntax().descendants() {
+                                if let Some(record_field) = ast::RecordField::cast(desc) {
+                                    vis_node_occurences.push(record_field.syntax().clone());
+                                }
+                            }
+                        },
+                        ast::TypeAlias(it) => {
+                            let node_def = {
+                                let nod = ctx.sema.to_def(&it)?;
+                                Definition::ModuleDef(nod.into())
+                            };
+
+                            self.get_and_process_usages(ctx, builder, node_def);
+                            vis_node_occurences.push(it.syntax().clone());
+                        },
+                        ast::Const(it) => {
+                            let node_def = {
+                                let nod = ctx.sema.to_def(&it)?;
+                                Definition::ModuleDef(nod.into())
+                            };
+
+                            self.get_and_process_usages(ctx, builder, node_def);
+                            vis_node_occurences.push(it.syntax().clone());
+                        },
+                        ast::Static(it) => {
+                            let node_def = {
+                                let nod = ctx.sema.to_def(&it)?;
+                                Definition::ModuleDef(nod.into())
+                            };
+
+                            self.get_and_process_usages(ctx, builder, node_def);
+                            vis_node_occurences.push(it.syntax().clone());
+                        },
+                        ast::Fn(it) => {
+                            let node_def = {
+                                let nod = ctx.sema.to_def(&it)?;
+                                Definition::ModuleDef(nod.into())
+                            };
+
+                            self.get_and_process_usages(ctx, builder, node_def);
+                            vis_node_occurences.push(it.syntax().clone());
+                        },
+                        ast::Impl(it) => {
+                            for desc in it.syntax().descendants() {
+                                if let Some(fn_) = ast::Fn::cast(desc) {
+                                    vis_node_occurences.push(fn_.syntax().clone());
+                                }
+                            }
+                        },
```

Comment:
This is missing a `get_and_process_usages` I believe? This also doesn't handle associated types and consts.